[ZEPPELIN-2287] Add more test to ensure 'RunOnSelectionChange' works well

AhyoungRyu opened the pull request:
### What is this PR for?
To address this feedback https://github.com/apache/zeppelin/pull/2100#issuecomment-286780437, I added more test cases to ensure `RunOnSelectionChange` works well.

The newly added test cases are like below. When we run this paragraph, one dropdown menu will be shown: 
```
%md 
My selection is ${my selection=1,1|2|3}
```
![screen shot 2017-03-20 at 4 00 20 pm](https://cloud.githubusercontent.com/assets/10060731/24090750/6a183e7c-0d86-11e7-8a66-60ed8008d2db.png)

1. When `RunOnSelectionChange` is **true** (default)
if we select "2" in the dropdown menu, the result text will be updated to "My selection is 2" right after the selection change. [See here](https://github.com/apache/zeppelin/compare/master...AhyoungRyu:ZEPPELIN-2287/addTestForRunOnSelectionChange?expand=1#diff-48409cf8ca5ddb1aea1a62ada3cd091dR282)

2. When `RunOnSelectionChange` is **false**
If we select "1" in the dropdown menu, the result text won't be updated to "My selection is 1" right after the selection change instead it'll be remained as "My selection is 2". [See here](https://github.com/apache/zeppelin/compare/master...AhyoungRyu:ZEPPELIN-2287/addTestForRunOnSelectionChange?expand=1#diff-48409cf8ca5ddb1aea1a62ada3cd091dR295)
But we can run the paragraph by pressing `Enter`. After running the paragraph manually using `Enter`, then the result text will be updated to "My selection is 1" [See here](https://github.com/apache/zeppelin/compare/master...AhyoungRyu:ZEPPELIN-2287/addTestForRunOnSelectionChange?expand=1#diff-48409cf8ca5ddb1aea1a62ada3cd091dR301)


### What type of PR is it?
Add test case 

### What is the Jira issue?
[ZEPPELIN-2287](https://issues.apache.org/jira/browse/ZEPPELIN-2287)

### How should this be tested?
After applying this patch, run 
```
TEST_SELENIUM="true" mvn package -DfailIfNoTests=false -pl 'zeppelin-interpreter,zeppelin-zengine,zeppelin-server' -Dtest=ParagraphActionsIT#testRunOnSelectionChange
```

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no

Leemoonsoo commented:
CI problem has fixed on master branch. Can you rebase and see if this PR passes the tests?

AhyoungRyu commented:
I rebased but failed again in Selenium test. All test cases that I added passed in my local but it's failed in here. Seem I need to do something more to avoid this..

AhyoungRyu commented:
All CI tests have passed now. Ready for review again :)

Leemoonsoo commented on zeppelin-server/src/test/java/org/apache/zeppelin/integration/ParagraphActionsIT.java:
```diff
@@ -257,31 +257,56 @@ public void testDisableParagraphRunButton() throws Exception {
+
+      // run paragraph manually by pressing ENTER
+      driver.findElement(By.xpath(xpathToDropdownMenu)).sendKeys(Keys.ENTER);
+      ZeppelinITUtils.sleep(1000, true);
```
Do you think we can use `waitForParagraph(1, "FINISHED");` instead of sleep?
'sleep' possibly make flaky test.

AhyoungRyu commented:
@Leemoonsoo I tried to replace `ZeppelinITUtils.sleep(1000, true)` with `waitForParagraph(1, "FINISHED")` but wan't enough and failed again. So I added  `driver.navigate().refresh()` before `waitForParagraph(1, "FINISHED")` and it passed.

Could you check again?

Leemoonsoo commented:
I think it should able to do without refresh().
Do you have idea why it was failed when `waitForParagraph(1, "FINISHED")` replaced sleep?

AhyoungRyu commented:
@Leemoonsoo Right it should be passed without refreshing. So I removed that part and added `waitForParagraph(1, "FINISHED");` before every selections for the stability. Now CI is green now. 
Will be appreciated if you can check again.

Leemoonsoo commented:
LGTM!

AhyoungRyu commented:
@Leemoonsoo Thanks! 
Merge into master and branch-0.7 if there are no more comments
